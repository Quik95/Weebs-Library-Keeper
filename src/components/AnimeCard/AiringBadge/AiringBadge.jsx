import React from "react";
import PropTypes from "prop-types";

//MUI
import Badge from "@material-ui/core/Badge";

//styles
import useStyles from "./AiringBadge.style";

//Custom components
import BadgeContent from "./BadgeContent/BadgeContent";

function AiringBadge({ airingIn, children }) {
  const classes = useStyles();

  return (
    <Badge
      invisible={!airingIn}
      badgeContent={<BadgeContent absTime="15/12/2019 18:29" relTime="9h" />}
      color="secondary"
      className={classes.gridFix}
    >
      {children}
    </Badge>
  );
}

AiringBadge.propTypes = {
  airingIn: PropTypes.number,
  children: PropTypes.element.isRequired
};

export default AiringBadge;
