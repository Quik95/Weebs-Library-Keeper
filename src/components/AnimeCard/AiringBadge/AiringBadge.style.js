import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  gridFix: { width: '100%' }
}));

