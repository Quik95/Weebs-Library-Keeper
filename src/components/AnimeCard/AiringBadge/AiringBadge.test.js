import React from 'react'

import AiringBadge from './AiringBadge'

describe('Renders AiringBadge', () => {
  const wrapper = shallow(<AiringBadge airingIn={12414}><div></div></AiringBadge>)

  it('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

})