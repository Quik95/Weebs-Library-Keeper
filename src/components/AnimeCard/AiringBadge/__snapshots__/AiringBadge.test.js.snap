// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders AiringBadge Renders correctly 1`] = `
<WithStyles(ForwardRef(Badge))
  badgeContent={
    <Memo(BadgeContent)
      absTime="15/12/2019 18:29"
      relTime="9h"
    />
  }
  className="makeStyles-gridFix-1"
  color="secondary"
  invisible={false}
>
  <div />
</WithStyles(ForwardRef(Badge))>
`;
